compiler/rustc_const_eval/src/const_eval/eval_queries.rs

```diff
@@ -26,12 +27,20 @@ const NOTE_ON_UNDEFINED_BEHAVIOR_ERROR: &str = "The rules on what exactly is und
      so this check might be overzealous. Please open an issue on the rustc \
      repository if you believe it should not be considered undefined behavior.";
 
+static SIGNALED: AtomicBool = AtomicBool::new(false);
+
 // Returns a pointer to where the result lives
 fn eval_body_using_ecx<'mir, 'tcx>(
     ecx: &mut CompileTimeEvalContext<'mir, 'tcx>,
     cid: GlobalId<'tcx>,
     body: &'mir mir::Body<'tcx>,
 ) -> InterpResult<'tcx, MPlaceTy<'tcx>> {
+    let _ = ctrlc::set_handler(move || {
```

Comment:
AFAICT, this will get silently skipped if any handler exists prior to it. Can we move this to rustc_driver or some other entry point and make signaled a global that's usable everywhere or so?